### PR TITLE
tests for insert and save of null record entity

### PR DESCRIPTION
**Pull request by njr-11**
Add tests for insert of a null record entity and save of a list of record entities containing a null entity.
Behavior should be a NullPointerException with a helpful message. We already have the NLS message written and used in other scenarios such as update or delete of null entity, but we are missing the code path for the other life cycle operations.
These test scenarios will also be helpful to get a record entity into the errorpaths bucket so that introspector output can be logged for it, although it will be a separate PR to implement that code path.

- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

### CODE REVIEW

**Comment by njr-11** [comment]
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=968063cb-b5ad-4473-a276-85ca2fc610fd) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

**Comment by njr-11** [comment]
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=5b5d7698-35d7-411e-86db-3098834ec1be) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

File: dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/QueryInfo.java [hunk 1]
```diff
@@ -4836,11 +4835,19 @@ final double toDouble(Object o) {
...
+        if (o == null)
+            throw exc(NullPointerException.class,
+                      "CWWKD1015.null.entity.param",
+                      method.getName(),
+                      repositoryInterface.getName());
+
         Object entity = o;
-        Class<?> oClass = o == null ? null : o.getClass();
+        Class<?> oClass = o.getClass();
         if (o != null && oClass.isRecord())
```
Comment (gjwatts) [inline dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/QueryInfo.java]:
```suggestion
        if (oClass.isRecord())
```

Comment (gjwatts) [reply dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/QueryInfo.java]: FYI, but not required for approval.

**Review by gjwatts — DISMISSED** [review]

**Review by gjwatts — APPROVED** [review]

**Comment by njr-11** [comment]
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=c1e41fb8-706b-4b6a-ab4a-de78263dadf6) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._
